logger.go

```diff
@@ -0,0 +1,84 @@
+package main
+
+import (
+	"code.cloudfoundry.org/lager"
+	"go.uber.org/zap"
+)
+
+// Ensure LagerZapLogger adheres to the Logger interface.
+var _ lager.Logger = &LagerZapLogger{}
+
+// LagerZapLogger is implementing the Lager interface. The OSBAPI expects us to use the lager logger,
+// but we wanted to use the Zap logger for its fast, leveled, and structured logging.
+// The zap methods are wrapped in the Lager method calls and is merely mapping them.
+type LagerZapLogger struct {
+	logger *zap.SugaredLogger
+}
+
+// NewLagerZapLogger constructs and returns a new LagerZapLogger with a pointer field of type SugaredLogger.
+func NewLagerZapLogger(zap *zap.SugaredLogger) *LagerZapLogger {
+	return &LagerZapLogger{
+		logger: zap,
+	}
+}
+
+// RegisterSink represents a write destination for a Logger. It provides a thread-safe interface for writing logs.
+func (lagerZapLogger *LagerZapLogger) RegisterSink(sink lager.Sink) {}
+
+// SessionName returns the name of the session. This is normally added when initializing a new logger but zap doesn't require nor need it.
+// but the OSBAPI does. Currently it's only returning an empty string.
+func (lagerZapLogger *LagerZapLogger) SessionName() string {
+	return ""
+}
+
+// Session sets the session of the logger and returns a new logger with a nested session. We are currently
+// returning the same logger back.
+func (lagerZapLogger *LagerZapLogger) Session(task string, data ...lager.Data) lager.Logger {
+	return lagerZapLogger
+}
+
+// WithData creates a new child with the parent fields and returns a logger with newly added data. We are currently
+// returing the same logger back.
```

Comment:
```suggestion
// returning the same logger back.
```

Reply:
I'll add this change. I presume seeing that these are mostly comments I can push directly to master?